src/main/java/ch/uzh/ifi/hase/soprafs24/service/PlayerService.java

```diff
@@ -25,29 +27,34 @@ public Player createPlayer(String playerName, String sessionId) {
         Player newPlayer = new Player();
         newPlayer.setName(playerName);
         newPlayer.setSessionId(sessionId);
-        // newPlayer.setOrderIndex == TODO
         newPlayer.setId(UUID.randomUUID().toString());
         newPlayer = playerRepository.save(newPlayer);
         playerRepository.flush();
         return newPlayer;
     }
 
     public void deletePlayer(String playerId) {
-        String cleanedPlayerId = cleanId(playerId);
-        playerRepository.deleteById(cleanedPlayerId);
+        playerRepository.deleteById(playerId);
     }
 
     public Player getPlayerById(String playerId) {
-        String cleanedPlayerId = cleanId(playerId);
-        return playerRepository.findById(cleanedPlayerId);
+        return playerRepository.findById(playerId);
     }
 
     public List<Player> getPlayersInSession(String sessionId) {
-        String cleanedSessionId = cleanId(sessionId);
-        return playerRepository.findAllBySessionId(cleanedSessionId);
+        return playerRepository.findAllBySessionId(sessionId);
     }
 
-    private String cleanId(String id) {
-        return id.replaceAll("^\"|\"$", "");
+    public void distributeOrderIndex(String sessionID){
+        List<Player> players = this.getPlayersInSession(sessionID);
+        int currentOrderIndex = 0;
+        for (Player player : players){
+            player.setOrderIndex(currentOrderIndex);
+            currentOrderIndex++;
+
+            playerRepository.save(player);
+            playerRepository.flush();
+        }
```

Comment:
Looks good but i would argue for a classical for loop with a i index and set this i to the orderindex... eg:
```
public void distributeOrderIndex(String sessionID) {
    List<Player> players = this.getPlayersInSession(sessionID);
    for (int i = 0; i < players.size(); i++) {
        Player player = players.get(i);
        player.setOrderIndex(i);
        playerRepository.save(player);
        playerRepository.flush();
    }
}
```